util/topgen/lib.py

```diff
@@ -566,3 +587,28 @@ def is_lc_ctrl(modules):
             return True
 
     return False
+
+
+def init_mmio_region(top, top_name, memories, devices):
+    """
+    Computes the bounds of the MMIO region.
+
+    MMIO region excludes any memory that is separate from the module configuration
+    space, i.e. ROM, main SRAM, and flash are excluded but retention SRAM,
+    spi_device memory, or usbdev memory are included.
+    """
+    regions = []
+    for ((dev_name, if_name), region) in devices:
+        module = get_module_by_name(top, dev_name)
+      
+        if module['base_addrs'][if_name].get("cpu_visible", True):
+            regions.append(region)
```

Comment:
Please no hard-coded address space name exceptions. This violates the data structure's layout.

Reply:
Exactly, I thought about that too. What about an address space validation as proposed in https://github.com/lowRISC/opentitan/pull/23274#discussion_r1618895737 ?